features/api/user.feature

```diff
@@ -10,6 +10,77 @@ Feature: User management
     Then the response status code should be 200
     And I should get a paged collection with the context "/api/contexts/User"
 
+#START -----Filter validation-----
+
+  Scenario: Filter users by type
+    When I send a GET request to "/api/users?filter[where][type]=contractor"
+    Then the response status code should be 200
+    And I should get a paged collection with the context "/api/contexts/User"
+    And the JSON node "hydra:totalItems" should be equal to 45
+    And the JSON node "types" of the objects of the JSON node "hydra:member" should contains "TYPE_CONTRACTOR"
```

Comment:
L20, in this step we want to test that all the users returned are contractors. To do that we have to:

* check for each user object of `hydra:member` that its type is `"TYPE_CONTRACTOR"
* as it is a paginated collection, if there is another page the check should be also done for the other pages

This cannot be achieved with a build-in step. So we must create a custom one in `ApiContext`, and for the sake of clarity, we name the step as: `And the users returned must be have the type "TYPE_CONTRACTOR"` or something like that